### ISSUE TITLE
spacemacs|dimish does not work when spaceline is excluded

### ISSUE THREAD

**Issue by sooheon** [syl20bnr/spacemacs#5194]
#### Description

Calls to `spacemacs|diminish` when spaceline is excluded from dotspacemacs is ineffectual. Is there some unnecessary coupling there?
#### Reproduction guide
- Add `spaceline` to `dotspacemacs-excluded-packages`
- Set `dotspacemacs-mode-line-unicode-symbols nil`
- Restart Emacs
- Observe non-diminished modeline
- Interactively (or in code) call something like `(spacemacs|diminish evil-cleverparens-mode nil " cp")`

_Observed behaviour:_
No diminishing!

_Expected behaviour:_
Diminishing according to given string.
#### System Info
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.105.11
- Spacemacs branch: develop (rev. fd93264)
- Distribution: spacemacs
- Editing style: hybrid
- Completion: ivy
- Layers:

``` elisp
((auto-completion :variables auto-completion-tab-key-behavior 'cycle auto-completion-return-key-behavior 'complete auto-completion-enable-snippets-in-popup t auto-completion-enable-sort-by-usage t)
 better-defaults colors clojure dash emacs-lisp
 (evil-snipe :variables evil-snipe-enable-alternate-f-and-t-behaviors t)
 finance html markdown
 (shell :variables shell-default-position 'bottom shell-default-shell 'eshell)
 git org osx pandoc rcirc syntax-checking spacemacs-ivy version-control
 (vinegar :variables vinegar-reuse-dired-buffer t)
 sooheon)
```

**Comment by TheBB** [comment]
Yes, this doesn't work if spaceline is not enabled.

**Comment by person808** [comment]
Seems like the function `spacemacs//prepare-diminish` is responsible

**Comment by justbur** [comment]
I just ran into this. Here's a workaround

```elisp
  (defun spacemacs/diminish-hook (_)
    (cl-loop for (mode uni nouni) in spacemacs--diminished-minor-modes
             do
             (diminish mode
                       (if dotspacemacs-mode-line-unicode-symbols
                           uni nouni))))
  (add-hook 'after-load-functions 'spacemacs/diminish-hook)
```

**Comment by TheBB** [comment]
Seems like a reasonable thing to put in `spacemacs-base/init-diminish` if `spaceline` is not present.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.

**Comment by lebensterben** [comment]
Currently, `spacemacs|diminish` is not working when `spaceline` is enabled...
which is opposite to the original issue...
